datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -312,7 +326,11 @@ impl Accumulator for SumAccumulator {
     fn evaluate(&self) -> Result<ScalarValue> {
         // TODO: add the checker for overflow
         // For the decimal(precision,_) data type, the absolute of value must be less than 10^precision.
-        Ok(self.sum.clone())
+        if self.count == 0 {
```

Comment:
It took me a while to understand why this `count` field is necessary. I am a little worried that it might add overhead to the aggregate function (aka slow down aggregates). However, we can always implement a special case sum for window functions if this is a problem. 

I think adding some comments saying that the count is needed to distinguish the case of "no values" if all values are  removed via `remove_batches` would help future readers. 

I did try commenting it out and there was a test diff, so 👍  for the coverage.